posthog/clickhouse/migrations/0135_session_replay_events_retention_period.py

```diff
@@ -0,0 +1,25 @@
+from posthog.clickhouse.client.migration_tools import run_sql_with_exceptions
+from posthog.session_recordings.sql.session_replay_event_migrations_sql import (
+    ADD_RETENTION_PERIOD_DISTRIBUTED_SESSION_REPLAY_EVENTS_TABLE_SQL,
+    ADD_RETENTION_PERIOD_SESSION_REPLAY_EVENTS_TABLE_SQL,
+    ADD_RETENTION_PERIOD_WRITABLE_SESSION_REPLAY_EVENTS_TABLE_SQL,
+    DROP_KAFKA_SESSION_REPLAY_EVENTS_TABLE_SQL,
+    DROP_SESSION_REPLAY_EVENTS_TABLE_MV_SQL,
+)
+from posthog.session_recordings.sql.session_replay_event_sql import (
+    KAFKA_SESSION_REPLAY_EVENTS_TABLE_SQL,
+    SESSION_REPLAY_EVENTS_TABLE_MV_SQL,
+)
+
+operations = [
+    # looking at past migrations we have to drop materialized views and kafka tables first
+    run_sql_with_exceptions(DROP_SESSION_REPLAY_EVENTS_TABLE_MV_SQL(on_cluster=False)),
+    run_sql_with_exceptions(DROP_KAFKA_SESSION_REPLAY_EVENTS_TABLE_SQL(on_cluster=False)),
+    # alter the target tables
+    run_sql_with_exceptions(ADD_RETENTION_PERIOD_WRITABLE_SESSION_REPLAY_EVENTS_TABLE_SQL()),
+    run_sql_with_exceptions(ADD_RETENTION_PERIOD_DISTRIBUTED_SESSION_REPLAY_EVENTS_TABLE_SQL()),
+    run_sql_with_exceptions(ADD_RETENTION_PERIOD_SESSION_REPLAY_EVENTS_TABLE_SQL()),
```

Comment:
Adding this to update the distributed table coordinator nodes in EU and only the sharded table once per shard!

```suggestion
    run_sql_with_exceptions(ADD_RETENTION_PERIOD_DISTRIBUTED_SESSION_REPLAY_EVENTS_TABLE_SQL(), node_role=NodeRole.ALL),
    run_sql_with_exceptions(ADD_RETENTION_PERIOD_SESSION_REPLAY_EVENTS_TABLE_SQL(), node_role=NodeRole.DATA, sharded=True),
```